src/test/test-copy.c

```diff
@@ -323,4 +323,48 @@ TEST(copy_proc) {
         assert_se(!isempty(a));
 }
 
+TEST_RET(copy_holes) {
+        char fn[] = "/var/tmp/test-copy-hole-fd-XXXXXX";
+        char fn_copy[] = "/var/tmp/test-copy-hole-fd-XXXXXX";
+        struct stat stat;
+        int r, fd, fd_copy;
+
+        fd = mkostemp_safe(fn);
+        assert_se(fd >= 0);
+
+        fd_copy = mkostemp_safe(fn_copy);
+        assert_se(fd >= 0);
```

Comment:
(doesn't really matter as this will fail on copy_bytes below anyway, but) this should be `assert_se(fd_copy >= 0)`

Reply:
Thanks. Fix is waiting in #22154.

Reply:
Woops, my bad, good catch!